hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServerConnection.java

```diff
@@ -184,26 +182,13 @@ public void close(String msg, Throwable cause) {
             return;
         }
 
-        if (localConnection != null) {
-            localConnection.close(msg, cause);
+        if (otherConnection != null) {
+            otherConnection.close(msg, cause);
         }
 
         if (lifecycleListener != null) {
             lifecycleListener.onConnectionClose(this, cause, false);
         }
-
-        if (localNodeEngine != null) {
```

Comment:
This cleanup was already performed in `onConnectionClose` (MockServer#L354), but since the other side of the connection is not closed properly due to this error, this cleanup was not performed in the opposite side member.